### PR TITLE
Banners - application & web layers

### PR DESCRIPTION
**Pull request by krknev**

### CODE REVIEW

**Review by yovko93 — APPROVED** [review]

File: CommonSystem.Application/BoundedContexts/Banners/Queries/ListPortalBannersByPortalQuery.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿namespace CommonSystem.Application.BoundedContexts.Banners.Queries
+{
+    #region Usings
+    using MediatR;
+    using CommonSystem.Application.Common;
+    using System.Threading.Tasks;
+    using System.Threading;
+    #endregion
+    public class ListPortalBannersByPortalQuery : IRequest<Result<List<ShortPortalBannersByPortalOutputModels>>>
```
Comment (Simeon-Yankov) [inline CommonSystem.Application/BoundedContexts/Banners/Queries/ListPortalBannersByPortalQuery.cs:10]: Maybe ICollection or IEnumerable

File: CommonSystem.Application/BoundedContexts/Banners/Queries/ListPortalBannersByPortalQuery.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            #region Methods
+            public Task<Result<List<ShortPortalBannersByPortalOutputModels>>> Handle(ListPortalBannersByPortalQuery request, CancellationToken cancellationToken)
+            {
+                throw new NotImplementedException();
+            }
+            #endregion
+        }
+    }
+
+    public class ShortPortalBannersByPortalOutputModels
```
[the remark this answers is not in the document]
Comment (krknev) [reply CommonSystem.Application/BoundedContexts/Banners/Queries/ListPortalBannersByPortalQuery.cs:31]: Where?

File: CommonSystem.Application/BoundedContexts/History/Commands/CreateHboHistoryCommand.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿namespace CommonSystem.Application.BoundedContexts.History.Commands
+{
+    using CommonSystem.Application.Common;
```
Comment (Simeon-Yankov) [inline CommonSystem.Application/BoundedContexts/History/Commands/CreateHboHistoryCommand.cs:3]: move in region

File: CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+{
+
+    #region USings
+    using MediatR;
+    using CommonSystem.Application.Common;
+    using CommonSystem.Application.Common.Pagination;
+    using CommonSystem.Domain.BoundedContexts.History.Hbo.Models;
+    #endregion
+
+    public class ListHboHistoriesQuery : IRequest<Result<ListHboHistoriesResult>>
```
Comment (Simeon-Yankov) [inline CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs]: Make all outputModel or all Result

File: CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        #region Props
+        public string UserId { get; set; } = default!;
+        public AppPaginationRequest Paging = new();
```
Comment (Simeon-Yankov) [inline CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs:16]: Same for request

File: CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            #region Methods
+            public Task<Result<ListHboHistoriesResult>> Handle(ListHboHistoriesQuery request, CancellationToken cancellationToken)
+            {
+                throw new NotImplementedException();
+            }
+            #endregion
+        }
+    }
+
+    public class ListHboHistoriesResult
```
Comment (Simeon-Yankov) [inline CommonSystem.Application/BoundedContexts/History/Queries/ListHboHistoriesQuery.cs]: extract & maybe rename

File: CommonSystem.Domain/BoundedContexts/Banners/Exceptions/PortalBannerException.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿namespace CommonSystem.Domain.BoundedContexts.Banners.Exceptions
+{
+
+    using CommonSystem.Domain.Common;
+
+    public class PortalBannerException : BaseDomainException
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/Banners/Exceptions/PortalBannerException.cs:6]: Rename to InvalidPortalBannerException

File: CommonSystem.Domain/BoundedContexts/Banners/Factories/PortalBannerFactory.cs [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+            {
+                throw new PortalBannerException($"{nameof(PortalBanner)}, {nameof(contentBgIsSet)} must have a value.");
+            }
+            if (!positionIsSet)
+            {
+                throw new PortalBannerException($"{nameof(PortalBanner)}, {nameof(positionIsSet)} must have a value.");
+            }
+            if (!portalIsSet)
+            {
+                throw new PortalBannerException($"{nameof(PortalBanner)},{nameof(portalIsSet)} must have a value.");
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/Banners/Factories/PortalBannerFactory.cs:52]: missing interval/space in exception text

File: CommonSystem.Domain/BoundedContexts/Banners/Models/PortalBanner.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+            Validation(createdFrom, contentBg, contentEng, position);
+            ContentBg = contentBg;
+            ContentEng = contentEng;
+            Portal = portal;
+            Position = position;
+        }
+        #endregion
+
+        #region Props
+        public string ContentBg { get; private set; } = default!;
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/Banners/Models/PortalBanner.cs:38]: No need of default

File: CommonSystem.Domain/BoundedContexts/Banners/Models/PortalBanner.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        #endregion
+
+        #region Validation
+        private void Validation(Guid createdFrom, string contentBg, string contentEng, string position)
+        {
+            ValidateCreatedFrom(createdFrom);
+            ValidateContent(contentBg, nameof(this.ContentBg));
+            ValidateContent(contentEng, nameof(this.ContentEng));
+            ValidateContent(position, nameof(this.Position));
+        }
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/Banners/Models/PortalBanner.cs:51]: Add new line after closing bracket  '}'

File: CommonSystem.Domain/BoundedContexts/History/Hbo/Exceptions/InvalidHboHistoryLogException.cs [hunk 11]
```diff
@@ -0,0 +1,17 @@
+﻿namespace CommonSystem.Domain.BoundedContexts.History.Hbo.Exceptions
+{
+
+    using CommonSystem.Domain.Common;
+
+    public class InvalidHboHistoryLogException : BaseDomainException
+    {
+        public InvalidHboHistoryLogException()
+        {
+                    }
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/History/Hbo/Exceptions/InvalidHboHistoryLogException.cs:10]: Format

File: CommonSystem.Domain/BoundedContexts/History/Hbo/Factories/HboHistoryLogsFactory.cs [hunk 12]
```diff
@@ -0,0 +1,77 @@
+﻿namespace CommonSystem.Domain.BoundedContexts.History.Hbo.Factories
+{
+    using CommonSystem.Domain.BoundedContexts.Banners.Exceptions;
+    using CommonSystem.Domain.BoundedContexts.History.Hbo.Exceptions;
```
Comment (Simeon-Yankov) [inline CommonSystem.Domain/BoundedContexts/History/Hbo/Factories/HboHistoryLogsFactory.cs:4]: usings out of region

**Review by Simeon-Yankov — CHANGES_REQUESTED** [review]

**Pull request merged**